webapp/channels/src/components/gif_picker/components/gif_picker_items.tsx

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useMemo} from 'react';
+import {GiphyFetch} from '@giphy/js-fetch-api';
+import {EmojiVariationsListProps, Grid} from '@giphy/react-components';
+
+const giphyFetch = new GiphyFetch('DxfEk2t4CR8bv4A1kviDBRMP9i1og3Da');
+
+const GUTTER_BETWEEN_GIFS = 8;
+const NUM_OF_GIFS_COLUMNS = 2;
+const GIF_DEFAULT_WIDTH = 350;
+const GIF_MARGIN_ENDS = 12;
+
+interface Props {
+    filter: string;
+    onClick: EmojiVariationsListProps['onGifClick'];
+}
+
+function GifPickerItems(props: Props) {
+    function fetch(offset: number) {
+        // We dont have to throttled the fetching as the library does it for us
+        if (props.filter.length > 0) {
+            return giphyFetch.search(props.filter, {offset, limit: 10});
+        }
+
+        return giphyFetch.trending({offset, limit: 10});
+    }
+
+    const width = useMemo(() => {
```

Comment:
Just to confirm, we are using `useMemo` because `getElementById` and `getBoundingClientrect` may be "expensive operations", because we don't want to change the value during different renders even if the elements in the body of the memo changes, or any other reason?

Reply:
yes and also https://github.com/mattermost/mattermost/pull/24236#discussion_r1295528494